mvpa2/generators/resampling.py

```diff
@@ -203,3 +205,92 @@ def generate(self, ds):
     def __str__(self):
         return _str(self, str(self._amount), n=self._attr, count=self.count,
                     apply_selection=self._apply_selection)
+
+
+class LogExclusions(Node):
+    """Log excluded entries
+
+    Given a dataset with a boolean sample or feature attribute, return a
+    dataset with only those samples/features marked `True`.
+    """
+    def __init__(self, fname, append=True, space='balanced_set', **kwargs):
+        """
+        Parameters
+        ----------
+        space : str
+          Name of the selection marker attribute in the input dataset that
+          indicates the desired subset.
+        """
+        Node.__init__(self, space=space, **kwargs)
+        self._fname = fname
+        # Truncate at start, append otherwise, to avoid holding an open
+        # filehandle throughout execution
+        if not append:
+            open(fname, 'w').close()
+
+    def _call(self, ds):
+        space = self.get_space()
+        _attr, collection = ds.get_attr(space)
+
+        battr = collection[space].value
+        if battr.dtype != np.dtype('bool'):
+            raise ValueError("LogExclusions space must be bools")
+
+        if collection is ds.sa:
+            excluded = ds[~battr]
+            desc = zip(excluded.sa.chunks, excluded.sa.targets,
+                       excluded.sa.time_coords)
+        elif collection is ds.fa:
+            desc = ds[:, ~battr].fa.voxel_indices
+        else:
+            # paranoid
+            raise RuntimeError(
+                    "Don't know where this collection comes from. "
+                    "This should never happen!")
+
+        with open(self._fname, 'a') as fobj:
+            fobj.write('# New entry {}\n'.format(time.ctime()))
+            for entry in desc:
+                fobj.write(','.join(map(str, entry)))
+                fobj.write('\n')
+
+        return ds
+
+
+class ApplySelection(Node):
+    """Applies a selection to a dataset
+
+    Given a dataset with a boolean sample or feature attribute, return a
+    dataset with only those samples/features marked `True`.
```

Comment:
this 2 doc lines seems to be way to similar to the previous class's ones... so what is the difference?
also ApplySelection sounds way too generic and could be of different kinds